Add comments on NetInfo.js

alvinthen opened the pull request:
Further describe the methods available on NetInfo.

Question: How do I update the docs on the website?

facebook-github-bot-1 commented:
By analyzing the blame information on this pull request, we identified @bestander and @satya164 to be potential reviewers.

facebook-github-bot-0 commented:
@alvinthen updated the pull request.

janicduplessis commented:
The docs on the website are generated from the code so the comments you added should show up on the website. You can test it locally with
```
cd website
npm install
npm start
```

janicduplessis commented on Libraries/Network/NetInfo.js:
```diff
@@ -168,6 +168,9 @@ const _isConnectedSubscriptions = new Map();
  */
 const NetInfo = {
+  /*
+   * Invokes the handler whenever the network info changes. The handler is passed with any of the connectivity types listed above.
```
Wrap the comments at around 80 characters.

alvinthen replied:
@janicduplessis from the contributing guide, it says

> Do not wrap lines at 80 characters - configure your editor to soft-wrap when editing documentation.

Did I understand incorrectly?

satya164 replied:
Can you change it to `Invokes the listener whenever network status changes. The listener receives one of the connectivity types listed above.`

janicduplessis replied:
This is for markdown files in the docs section. We  wrap doc blocks at ~80 for js files.

janicduplessis commented on Libraries/Network/NetInfo.js:
```diff
@@ -196,11 +202,18 @@ const NetInfo = {
     return RCTNetInfo.getCurrentConnectivity().then(resp => resp.network_info);
   },
 
   isConnected: {
```
Those won't show on the website as it doesn't extract methods from a nested object :( Still good to have the comments.

satya164 replied:
If it's not gonna show up on the website, we shouldn't add a comment here, as its obvious by looking at the method signature. We should update Flow types though.

janicduplessis commented on Libraries/Network/NetInfo.js:
````diff
@@ -168,6 +168,9 @@ const _isConnectedSubscriptions = new Map();
  * ```
  */
 const NetInfo = {
+  /*
````
Use doc blocks
```
/**
 *
 */
```
Pretty much just add one `*` to each block first line.

satya164 commented on Libraries/Network/NetInfo.js:
```diff
@@ -184,6 +187,9 @@ const NetInfo = {
   },
 
+  /*
+   * Stop listening to network info changes.
```
Can you change it to `Removes the listener for network status changes`

alvinthen commented:
I have updated as per requested. I moved the documentation of the methods of `isConnected` to the object itself, this can help developer to understand without going through the source code

facebook-github-bot-9 commented:
@alvinthen updated the pull request.

janicduplessis commented on Libraries/Network/NetInfo.js:
```diff
@@ -196,10 +203,19 @@ const NetInfo = {
   },
 
+  /**
+   * Returns a promise that returns one of the connectivity types listed above.
```
Let's change this to `Returns a promise that resolves with one ...`

janicduplessis commented on Libraries/Network/NetInfo.js:
```diff
@@ -196,10 +203,19 @@ const NetInfo = {
   },
 
+  /**
+   * An object with similar methods above but the listener receives a boolean
```
`An object with the same methods as above ...`

janicduplessis commented:
Can you just do those small wording changes then this is good to go, thanks!

facebook-github-bot-3 commented:
@alvinthen updated the pull request.

janicduplessis commented:
@facebook-github-bot shipit
